packages/vue-apollo-composable/src/useApolloClient.ts

```diff
@@ -37,9 +37,12 @@ export function useApolloClient<TCacheShape = any>(clientId?: ClientId): UseApol
   const providedApolloClients: ClientDict<TCacheShape> = inject(ApolloClients, null)
   const providedApolloClient: ApolloClient<TCacheShape> = inject(DefaultApolloClient, null)
 
-  function resolveClient() {
+  function resolveClient(resolvedClientId?: ClientId) {
+    if (resolvedClientId) {
+      clientId = resolvedClientId
```

Comment:
Won't this reassign this value for *all* subsequent requests? I think there needs to be a local variable here:

```javascript
const id = resolvedClientId ? resolvedClientId : clientId
```